### PR TITLE
jgrss/cleanup

### PR DESCRIPTION
**Pull request by jgrss**
## What is this PR changing?
Fixes bug in polygon extraction

### What was the issue?
Polygon conversion to points were based on the image affine and not the polygon feature affine, causing a one pixel shift north and west of the polygon top left bounds.

### CODE REVIEW

File: src/geowombat/core/util.py [hunk 1]
```diff
@@ -487,70 +527,65 @@ def sample_feature(df_row, id_column, df_columns, crs, res, all_touched, meta, f
...
-    y_samples = valid_samples[0] + int(round(abs(meta.top - geom_info.top)) / res[1])
-    x_samples = valid_samples[1] + int(round(abs(geom_info.left - meta.left)) / res[0])
-
-    # Convert the map indices to map coordinates
-    x_coords, y_coords = meta.affine * (x_samples, y_samples)
+    # Get geometry indices
+    x_samples = valid_samples[1]
+    y_samples = valid_samples[0]
+    # Convert the geometry indices to geometry map coordinates
+    x_coords, y_coords = geom_info.affine * (x_samples, y_samples)
```
Comment (jgrss) [inline src/geowombat/core/util.py:551]: This is the fix, part 1. The previous affine was the image affine and the replacement is the geometry affine.

File: src/geowombat/core/util.py [hunk 2]
```diff
@@ -487,70 +527,65 @@ def sample_feature(df_row, id_column, df_columns, crs, res, all_touched, meta, f
...
+    # Move coordinates offset from polygon left and top edges
+    x_coords += abs(res[1]) * 0.5
```
Comment (jgrss) [inline src/geowombat/core/util.py:553]: This is the fix, part 2. This moves points off of the polygon edges.

File: tests/test_extract.py [hunk 3]
```diff
@@ -50,12 +58,66 @@ def test_multi_image_multi_band(self):
...
+        )
         self.assertTrue(np.allclose(df.t1_red.values, l8_224078_20200518_values[2, :]))
         self.assertTrue(np.allclose(df.t2_blue.values, l8_224078_20200518_values[0, :]))
-        self.assertTrue(np.allclose(df.t2_green.values, l8_224078_20200518_values[1, :]))
+        self.assertTrue(
+            np.allclose(df.t2_green.values, l8_224078_20200518_values[1, :])
+        )
         self.assertTrue(np.allclose(df.t2_red.values, l8_224078_20200518_values[2, :]))
 
+    def test_points(self):
```
Comment (jgrss) [inline tests/test_extract.py:71]: New tests for point extraction

**Review by mmann1123 — APPROVED** [review]
Nice

**Pull request merged**